bitcoin/src/blockdata/script/borrowed.rs

```diff
@@ -489,6 +503,16 @@ impl Script {
         }
     }
 
+    /// Iterates the script to find the last pushdata.
+    ///
+    /// Returns `None` is the instruction is an opcode or if the script is empty.
+    pub(crate) fn last_pushdata(&self) -> Option<&PushBytes> {
+        match self.instructions().last() {
+            Some(Ok(Instruction::PushBytes(bytes))) => Some(bytes),
+            _ => None,
+        }
+    }
```

Comment:
In 862e0f4349f308a93f0de7caf47116d577e15fdb:

Similarly I think this needs to count "pushes" that include the numeric opcodes, at the very least, which may be nontrivial to fit into our type system. It also ought to count OP_RESERVED though this is less essential since such a script will always fail anyway.

Reply:
Perhaps it should just return `PushBytes::empty()` when an opcode <= OP_16 is present.

Reply:
@junderw well, the effect of a numeric opcode *is* to push a single byte number. So to be correct it should return some sort of `PushBytes` object containing the number. Maybe we could create a set of 16 static ones (-1 and 1 through 16) (zero is `PushBytes::empty`)?

Reply:
Since it's `pub(crate)` anyways I just added a simple enum to encapsulate the different states.